visualization/src/point_picking_event.cpp

```diff
@@ -167,49 +189,67 @@ pcl::visualization::PointPickingCallback::performSinglePick (
 /////////////////////////////////////////////////////////////////////////////////////////////
 int
 pcl::visualization::PointPickingCallback::performAreaPick (vtkRenderWindowInteractor *iren,
-                                                           std::vector<int> &indices)
+                                                           std::vector<std::vector<int> > &indices)
 {
   vtkAreaPicker *picker = static_cast<vtkAreaPicker*> (iren->GetPicker ());
   vtkRenderer *ren = iren->FindPokedRenderer (iren->GetEventPosition ()[0], iren->GetEventPosition ()[1]);
   picker->AreaPick (x_, y_, iren->GetEventPosition ()[0], iren->GetEventPosition ()[1], ren);
-  if (picker->GetDataSet ())
+ if (picker->GetProp3Ds())
   {
-    vtkPolyData* points = reinterpret_cast<vtkPolyData*> (picker->GetDataSet ());
-
-    // This is a naive solution till we fugure out where to add the GlobalIds at an earlier stage
-    if (!points->GetPointData ()->GetGlobalIds () ||
-        points->GetPointData ()->GetGlobalIds ()->GetNumberOfTuples () == 0)
+    int pt_numb=0;
+    vtkProp3DCollection* props = picker->GetProp3Ds();
+    vtkCollectionSimpleIterator pit;
+    vtkProp3D* prop;
+    for(props->InitTraversal(pit);(prop = props->GetNextProp3D(pit));)
     {
-      vtkSmartPointer<vtkIdTypeArray> global_ids = vtkIdTypeArray::New ();
-      global_ids->SetNumberOfValues (picker->GetDataSet ()->GetNumberOfPoints ());
-      for (vtkIdType i = 0; i < picker->GetDataSet ()->GetNumberOfPoints (); ++i)
-        global_ids->SetValue (i,i);
+      vtkActor* actor = vtkActor::SafeDownCast(prop);
+      if(actor)
```

Comment:
Outdent this entire block using `if (!actor) { continue; }`

Reply:
done